src/Channel.php

```diff
@@ -111,6 +111,18 @@ public function __construct(private Connection $connection, private Client $clie
 
     public function getClient(): Connection
```

Comment:
Since this method is only used by methods pulled in via a trait, what about we make it protected since there is no need for it to be public?

```suggestion
    protected function getClient(): Connection
```

Reply:
Sure, making that method protected sounds good, so I've done it.

I'm so sorry, but I really have no idea how to write test, and I wouldn't even know how to run the existing ones. I won't be able to help in this area.

Reply:
> Sure, making that method protected sounds good, so I've done it.

👍 

> I'm so sorry, but I really have no idea how to write test

Do you want to learn? Quite literally the code I put in the inital review should work.

> and I wouldn't even know how to run the existing ones. I won't be able to help in this area.

There is a section in the readme about how to run them: https://github.com/jakubkulhan/bunny?tab=readme-ov-file#testing

But in short (assuming Linux or a Mac): `make unit-testing` will spin up a rabbitmq using docker compose and run the tests aginst it.

Reply:
Sorry, but I’ve really never deal with this in my life. With zero experience I would probably cause more harm than good. I’d really like to help, but I’m a hobbyist, I’ve never worked in a corporate environment, and things like tests, docker, CI workflows, or this "phpcs" thing are simply beyond me and very demotivating. Maybe it’ll be better to simply reject my PRs and treat them as inspiration for implementing similar changes your own way.

Reply:
> beyond me and very demotivating

Ok lots of things to unpack, but let's start with this one: We/I do not want to demotivate you, and I'm more then happy to help and teach you those things. If you are open to it, and have the time and resources for those. If you can't run docker, we can focus on another part you might want to learn.

> Maybe it’ll be better to simply reject my PRs and treat them as inspiration for implementing similar changes your own way.

If this is what you prefer, that is also fine by me. But do know I'm offering to take the time sharing my knowledge and teach you.

> I’d really like to help, but I’m a hobbyist

Awesome! Do know that's how I started 20 years ago messing around, learning new things, getting frustrated and demotivated when things didn't work out, but also getting excited when they did.

> Sorry, but I’ve really never deal with this in my life. With zero experience I would probably cause more harm than good, 

Well you can't merge, and I'll be checking everything, plus testing things out in my own environment that will benefit from your PR so I do want them in. You will do good, not harm. And if things go wrong that is on me for not validating the changes properly. Don't worry about that.

> I’ve never worked in a corporate environment

Most of this tooling is found in a lot of open source projects, things in my personal namespace have more and are stricter. But the most important thing are tests.

> things like tests, docker, CI workflows, or this "phpcs"

Which of those would you like to learn? My suggestion would be tests as they are the validation and insurance the code works as expected and you don't need to learn any new tech. As long as you have RabbitMQ running somewhere to run the tests against.

Reply:
@appkit-framework TL;DR for my message above: If there is anything you want to learn, like testing, CI, docker, etc. I'll take care of the other aspects of the PR and will focus on teaching you the specific part you want to learn. As I see tremendous value in learning and I can take care of the rest so you have a place to learn and make you feel more comfortable.

Reply:
Sorry, but I can’t learn these things just for the sake of helping with your project. I’m doing perfectly fine without them, and I wouldn’t use them in my own work anyway, which I need to focus on right now.

Reply:
> Sorry, but I can’t learn these things just for the sake of helping with your project. I’m doing perfectly fine without them, and I wouldn’t use them in my own work anyway, which I need to focus on right now.

Don't be sorry about that, I would only want to teach you if you want to learn that for yourself so you can apply them in your own work. Thank you for your honesty, I really appriciate that, if at some point you do want to start learning one of more of these things you know where to find me. 

I'll take over your PR's from here then, add testing etc. So please keep them open and I'll get them in. Thank you for your work so far!